disassemblers/ofrak_pyghidra/src/ofrak_pyghidra/standalone/pyghidra_analysis.py

```diff
@@ -17,107 +17,120 @@ def _parse_offset(java_object):
-        from java.math import BigInteger
-
-        # If base_address is provided, rebase the program
-        if base_address is not None:
-            # Convert base_address to int if it's a string
-            if isinstance(base_address, str):
-                if base_address.startswith("0x"):
-                    base_address = int(base_address, 16)
-                else:
-                    base_address = int(base_address)
-
-            # Rebase the program to the specified base address
-            program = flat_api.getCurrentProgram()
-            address_factory = program.getAddressFactory()
-            new_base_addr = address_factory.getDefaultAddressSpace().getAddress(hex(base_address))
-            program.setImageBase(new_base_addr, True)
-
-        main_dictionary = {}
-        code_regions = _unpack_program(flat_api)
-        main_dictionary["metadata"] = {}
-        main_dictionary["metadata"]["backend"] = "ghidra"
-        main_dictionary["metadata"]["decompiled"] = decompiled
-        main_dictionary["metadata"]["path"] = program_file
-        if base_address is not None:
-            main_dictionary["metadata"]["base_address"] = base_address
-        with open(program_file, "rb") as fh:
-            data = fh.read()
-            md5_hash = hashlib.md5(data)
-            main_dictionary["metadata"]["hash"] = md5_hash.digest().hex()
-
-        for code_region in code_regions:
-            seg_key = f"seg_{code_region['virtual_address']}"
-            main_dictionary[seg_key] = code_region
-            func_cbs = _unpack_code_region(code_region, flat_api)
-            code_region["children"] = []
-
-            decomp_interface = DecompInterface()
-            prog_options = DecompileOptions()
-            prog_options.grabFromProgram(flat_api.getCurrentProgram())
-            decomp_interface.setOptions(prog_options)
-            init = decomp_interface.openProgram(flat_api.getCurrentProgram())
-            if not init:
-                raise RuntimeError("Could not open program for decompilation")
-
-            for func, cb in func_cbs:
-                cb_key = f"func_{cb['virtual_address']}"
-                code_region["children"].append(cb_key)
-                if decompiled:
-                    try:
-                        decompilation = _decompile(func, decomp_interface, TaskMonitor.DUMMY)
-                    except Exception as e:
-                        print(e, traceback.format_exc())
-                        decompilation = ""
-                    cb["decompilation"] = decompilation
-                bb_model = BasicBlockModel(flat_api.getCurrentProgram())
-                basic_blocks, data_words = _unpack_complex_block(
-                    func, flat_api, bb_model, BigInteger.ONE
+    try:
+        with pyghidra.open_program(program_file, language=language) as flat_api:
+            # Java packages must be imported after pyghidra.start or pyghidra.open_program
```

Comment:
Not worth blocking this PR for this, but I wonder if this function can be broken up into a couple of smaller parts? I think that might make it a bit easier to read without adding too much indirection. If there is no clear place to break the function apart, that's okay too – feel free to resolve this in that case.

Reply:
See #667